src/coreclr/src/jit/emit.cpp

```diff
@@ -5515,47 +5515,49 @@ UNATIVE_OFFSET emitter::emitFindOffset(insGroup* ig, unsigned insNum)
  *  block.
  */
 
-UNATIVE_OFFSET emitter::emitDataGenBeg(UNATIVE_OFFSET size, UNATIVE_OFFSET alignment)
+UNATIVE_OFFSET emitter::emitDataGenBeg(unsigned size, unsigned alignment, var_types dataType)
 {
     unsigned     secOffs;
     dataSection* secDesc;
 
     assert(emitDataSecCur == nullptr);
 
-    // The size must not be zero and must be a multiple of 4 bytes
-    // Additionally, 4 bytes is the minimum alignment that will
+    // The size must not be zero and must be a multiple of MIN_DATA_ALIGN
+    // Additionally, MIN_DATA_ALIGN is the minimum alignment that will
     // actually be used. That is, if the user requests an alignment
-    // of 1 or 2, they will get  something that is at least 4-byte
-    // aligned. We allow the others since 4 is at least 1/2 and its
-    // simpler to allow it than to check and block.
-    assert((size != 0) && ((size % 4) == 0));
+    // less than MIN_DATA_ALIGN, they will get  something that is at least
+    // MIN_DATA_ALIGN. We allow smaller alignment to be specified since it is
+    // simpler to allow it than to check and block itr.
+    //
+    assert((size != 0) && ((size % dataSection::MIN_DATA_ALIGN) == 0));
 
-    // This restricts the alignment to: 1, 2, 4, 8, 16, or 32 bytes
-    // Alignments greater than 32 would require VM support in ICorJitInfo::allocMem
+    // This restricts the alignment to MAX_DATA_ALIGN
+    // Alignments greater than 32 also require VM support in ICorJitInfo::allocMem
```

Comment:
```suggestion
    // Alignments greater than MAX_DATA_ALIGN also require VM support in ICorJitInfo::allocMem
```

Reply:
Actually you have add support for alignments greater than 32 to  ICorJitInfo::allocMem, before you can increase the emitter's MAX_DATA_ALIGN.   If we ever do that we may also move the definitions of MAX/MIN_DATA_ALIGN out of emit.h and into compiler.h

Reply:
Should the comment possibly be moved to be on `MAX_DATA_ALIGN` and indicate something like "this can't be increased without also touching `ICorJitInfo::allocMem`"